File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int m = -1;

        vector<int> result(arr.size());
        for(int x = arr.size() - 1; x >= 0; x--){
            result[x] = m;
            m = max(arr[x], m);
        }

        return result;
    }
};

File: Data Structures & Algorithms/string-matching-in-an-array/submission-2.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        result: set[str] = set()
        words = sorted(words, key=lambda x: len(x))
        for index in range(len(words)):
            for innerIndex in range(index + 1, len(words)):
                if words[innerIndex].find(words[index]) != -1:
                    result.add(words[index])

        return list(result)